core/corehttp/hostname.go

```diff
@@ -266,6 +290,36 @@ func isPeerIDNamespace(ns string) bool {
 	}
 }
 
+// Converts an identifier to DNS-safe representation
+func toDNSSafePrefix(id string) (prefix string) {
```

Comment:
I have a better version of this but something is wrong with my setup - tests won't run.
I'll push a PR against this branch when it's ready

Reply:
Thank you, 45e3271a0dad2a3438aacc56513f771ee8d9679d is now part of this PR.